lib/build.ml

```diff
@@ -148,7 +148,7 @@ module Make (Raw_store : S.STORE) (Sandbox : S.SANDBOX) (Fetch : S.FETCHER) = st
       (* Fmt.pr "COPY: %a@." Sexplib.Sexp.pp_hum (sexp_of_copy_details details); *)
       let id = Sha256.to_hex (Sha256.string (Sexplib.Sexp.to_string (sexp_of_copy_details details))) in
       Store.build t.store ?switch ~base ~id ~log (fun ~cancelled ~log result_tmp ->
-          let argv = ["tar"; "-xf"; "-"] in
+          let argv = Option.value ~default:(["tar"; "-xf"; "-"]) (Sandbox.tar t.sandbox) in
```

Comment:
IIUC, the new `tar` value added to the `S.SANDBOX` interface is working like this:

All `SANDBOX` implementations, except for the new QEMU one, define this value as 

```ocaml
let tar _ = None
```

and in the QEMU sandbox we get it as

```ocaml
let tar t =
  match t.qemu_guest_os with
  | Linux -> None
  | OpenBSD -> Some ["gtar"; "-xf"; "-"]
  | Windows -> Some ["/cygdrive/c/Windows/System32/tar.exe"; "-xf"; "-"; "-C"; "/"]
```

And this is the only place where this new `tar` gets used.

So, the reason it is a partial function `t -> string list option` is that the possible `None` value is used here to fallback to the default tar invocation. This suggests to me that `tar` should in fact be a total function `t -> string list`, and that default `let tar _ = ["tar"; "-xf"; "-"]` should instead be shared by all the implementations (e.g., via an `include` or reference to a `default_tar` value).

One immediate advantage of my suggestion is that anyone looking at a `SANDBOX` implementation would be able to see how tar is defined just by following the reference. As it is, there is no way to see what the actual value of the tar command is unless one finds where `Sandgox.tar` is being used, at which point one discovers what the `None` actually means.

If we end up needing to use `Sandbox.tar` somewhere else, having the implementation actually carry the command enables that without any fuss. Otherwise, we will need to duplicate this defaulting logic.

Under my suggestion, this code would then be

```suggestion
          let argv = Sandbox.tar t.sandbox in
```

enabled by the following suggested changes:

Reply:
I've incorporated some elements of this in the updated version.

Reply:
Looks good!